### PR TITLE
fix(react-components): search hook fix for instances with multiple connections to other 3d mapped instances

### PR DESCRIPTION
**Pull request by Savokr**
#### Type of change

Bug 

#### Jira ticket :blue_book:

https://cognitedata.atlassian.net/browse/BND3D-3522

## Description :pencil:

Fixes an issue when "second-level" mapped 3d instances were not always appearing in search results.

## How has this been tested? :mag:

In FDX and searchHooks story.

## Test instructions :information_source:

1. Go to searchHooks story 
2. Search for "work-order"
3. Observe an item with the type "WorkOrderMultiple" in the search results. It was not there before the fix.

## Checklist :ballot_box_with_check:

- [x] I am proud of this feature.
- [x] I have performed a self-review of my own code.
- [x] I have added PR type (Feat, Bug, Chore, Test, Docs, Style, Refactor) to the PR title.
- [x] I have manually tested this for different scenarios (different models, formats, devices, browsers).
- [x] I have commented my code in hard-to-understand areas.
- [x] I have made corresponding changes to the public documentation.
- [x] I have added unit and visuals tests to prove that my feature works to the best of my ability.
- [x] I have refactored the code for readability to the best of my ability.
- [x] I have checked that my changes do not introduce regressions in the public documentation.
- [x] I have outlined any known defects / lacking capabilities in the contents of this PR.
- [x] I have listed any remaining work that should follow this PR in the description or jira/miro/etc.
- [x] I have added TSDoc to any public facing changes.
- [x] I have added "developer documentation" in any package README.md that is related / applicable to this PR.
- [x] I have noted down and am currently tracking any technical debt introduced in this PR.
- [x] I have thoroughly thought about the architecture of this implementation.
- [ ] I have asked peers to test this feature.

### CODE REVIEW

File: react-components/src/hooks/useSearchMappedEquipmentFDM.tsx [hunk 1]
```diff
@@ -64,13 +65,22 @@ export const useSearchMappedEquipmentFDM = (
...
 
-        return transformedResults;
+        const combinedWithOtherViews = viewsToSearch.map((view) => ({
+          view,
+          instances:
+            transformedResults.find(
+              (result) =>
+                result.view.externalId === view.externalId && result.view.space === view.space
+            )?.instances ?? []
+        }));
```
Comment (haakonflatval-cognite) [inline react-components/src/hooks/useSearchMappedEquipmentFDM.tsx:75]: Just to understand - this makes sure that all views are associated with some results (if any), even if there are duplicate views in the `viewsToSearch` list?

File: react-components/src/utilities/FdmSDK.ts [hunk 2]
```diff
@@ -8,7 +8,7 @@ import { type FdmPropertyType } from '../components/Reveal3DResources/types';
 type InstanceType = 'node' | 'edge';
 type EdgeDirection = 'source' | 'destination';
 
-type InstanceFilter = any;
+export type InstanceFilter = any;
```
Comment (haakonflatval-cognite) [inline react-components/src/utilities/FdmSDK.ts:11]: Is this used anywhere? Should it be used in a parameter in `useSearchMappedEquipmentFDM`?

**Review by haakonflatval-cognite — APPROVED** [review]
LGTM :+1:

File: react-components/src/hooks/useSearchMappedEquipmentFDM.tsx (hunk 1 above)
Comment (Savokr) [reply react-components/src/hooks/useSearchMappedEquipmentFDM.tsx:75]: It just ensures that it returns "3d searchable" views even if there are no results for them

File: react-components/src/utilities/FdmSDK.ts (hunk 2 above)
Comment (Savokr) [reply react-components/src/utilities/FdmSDK.ts:11]: Was used before, removed export

**Pull request merged**